testsuite/integration/basic/src/test/java/org/wildfly/test/integration/observability/micrometer/BasicMicrometerTestCase.java

```diff
@@ -0,0 +1,151 @@
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+//@ServerSetup(MicrometerSetupTask.class)
+public class BasicMicrometerTestCase {
+    @Inject
+    private MeterRegistry meterRegistry;
+
+    @ArquillianResource
+    private URL url;
+    private Client client = ClientBuilder.newClient().register(JacksonFeature.class);
+    private ManagementClient managementClient;
+
+    private static final String WEB_XML =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                    + "\n"
+                    + "<web-app xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://xmlns.jcp.org/xml/ns/javaee\"\n"
+                    + "         xsi:schemaLocation=\"http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd\" version=\"4.0\">\n"
+                    + "    <servlet-mapping>\n"
+                    + "        <servlet-name>jakarta.ws.rs.core.Application</servlet-name>\n"
+                    + "        <url-pattern>/*</url-pattern>\n"
+                    + "    </servlet-mapping>"
+                    + "</web-app>\n";
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, BasicMicrometerTestCase.class.getSimpleName() + ".war")
+                .addClasses(MicrometerSetupTask.class, ServerSetupTask.class, MetricResource.class)
+                .addAsWebInfResource(new StringAsset(WEB_XML), "web.xml")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(createPermissionsXmlAsset(
+                        // Required for the ClientBuilder.newBuilder() so the ServiceLoader will work
+                        new FilePermission("<<ALL FILES>>", "read"),
+                        // Required for com.fasterxml.jackson.jaxrs.json.JacksonJaxbJsonProvider. During <init> there is a
+                        // reflection test to check for JAXRS 2.0.
+                        new RuntimePermission("accessDeclaredMembers"),
+                        // Required for the client to connect
+                        new SocketPermission(TestSuiteEnvironment.getHttpAddress() + ":" +
+                                TestSuiteEnvironment.getHttpPort(), "connect,resolve"),
+                        new SecurityPermission("insertProvider"),
+                        new MBeanServerPermission("createMBeanServer"),
+                        new MBeanPermission("*", "registerMBean, unregisterMBean, invoke"),
+                        new RuntimePermission("getClassLoader"),
+                        new RuntimePermission("modifyThread"),
+                        new RuntimePermission("setContextClassLoader")
+                ), "permissions.xml");
+    }
+
+    @Test
+    public void hasInjectedMeterRegistry() {
+        Assert.assertNotNull(meterRegistry);
+    }
```

Comment:
I think you should just add a GET method to MetricsResource and check this remotely. Make the entire class @RunAsClient.

I mostly say this because I'm not comfortable with all the things you need to add to permissions.xml -- they may obscure problems with the subsystem implementation by giving the deployment perms it wouldn't need otherwise.

Reply:
Fixed